cudax/include/cuda/experimental/__async/sender/run_loop.cuh

```diff
@@ -187,93 +185,53 @@ public:
   public:
     using scheduler_concept = scheduler_t;
 
-    [[nodiscard]] _CUDAX_API auto schedule() const noexcept -> __schedule_task
+    [[nodiscard]] _CUDAX_API auto schedule() const noexcept -> __sndr_t
     {
-      return __schedule_task{__loop_};
+      return __sndr_t{__loop_};
     }
 
-    _CUDAX_API friend bool operator==(const __scheduler& __a, const __scheduler& __b) noexcept
+    [[nodiscard]] _CUDAX_API friend bool operator==(const scheduler& __a, const scheduler& __b) noexcept
     {
       return __a.__loop_ == __b.__loop_;
     }
 
-    _CUDAX_API friend bool operator!=(const __scheduler& __a, const __scheduler& __b) noexcept
+    [[nodiscard]] _CUDAX_API friend bool operator!=(const scheduler& __a, const scheduler& __b) noexcept
     {
       return __a.__loop_ != __b.__loop_;
     }
   };
 
-  _CUDAX_API auto get_scheduler() noexcept -> __scheduler
+  [[nodiscard]] _CUDAX_API auto get_scheduler() noexcept -> scheduler
   {
-    return __scheduler{this};
+    return scheduler{this};
   }
 
-  _CUDAX_API void run();
-
-  _CUDAX_API void finish();
-
 private:
-  _CUDAX_API void __push_back(__task* __tsk);
-  _CUDAX_API auto __pop_front() -> __task*;
-
-  ::std::mutex __mutex{};
-  ::std::condition_variable __cv{};
-  __task __head{};
-  bool __stop = false;
-};
-
-template <class _Rcvr>
-_CUDAX_API inline void __operation<_Rcvr>::start() & noexcept {
-  _CUDAX_TRY(                                                               //
-    ({                                                                      //
-      __loop_->__push_back(this);                                           //
-    }),                                                                     //
-    _CUDAX_CATCH(...)                                                       //
-    ({                                                                      //
-      set_error(static_cast<_Rcvr&&>(__rcvr_), ::std::current_exception()); //
-    })                                                                      //
-    )                                                                       //
-}
-
-_CUDAX_API inline void run_loop::run()
-{
-  for (__task* __tsk = __pop_front(); __tsk != &__head; __tsk = __pop_front())
+  // Returns true if any tasks were executed.
+  _CUDAX_API bool __execute_all(bool __wait) noexcept
   {
-    __tsk->__execute();
+    // Wait until the queue has tasks to execute and then dequeue all of them.
+    auto __queue = __queue_.pop_all(__wait);
```

Comment:
That operation is not cromulent with the description of `__execute_all`. I would have expected to have a simple `false` return in case of an empty queue